frontend/src/components/LoginForm.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Alert, Button, FormControl, FormLabel, TextField } from "@mui/material";
+import { getAuth, signInWithEmailAndPassword } from "firebase/auth";
+
+const LoginForm = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [loginSuccessful, setLoginSuccessful] = useState(false);
+    const [loginErrorMessage, setLoginErrorMessage] = useState("");
+
+    const navigate = useNavigate();
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        const auth = getAuth();
+        signInWithEmailAndPassword(auth, email, password)
+            .then((userCredential) => {
+                setLoginSuccessful(true);
+                // The user is successfully signed in
+                setTimeout(() => {
+                    setLoginSuccessful(false);
+                    navigate("/profile");
+                }, 2000);
+            })
+            .catch((error) => {
+                switch (error.code) {
+                    case "auth/wrong-password":
+                        setLoginErrorMessage("Incorrect password");
+                        setTimeout(() => {
+                            setLoginErrorMessage("");
+                        }, 2000);
+                        break;
+                    case "auth/user-not-found":
+                        setLoginErrorMessage("User not found");
```

Comment:
"User" is quite a technical term. We should probably word this something like "Email does not have an account associated with it".

Sidenote: I don't like how Firebase exposes the email like this -- it allows an arbitrary user to see if someone has signed up with a certain email account.